pkg/agent/flowexporter/exporter.go

```diff
@@ -115,270 +145,277 @@ func NewFlowExporter(podStore objectstore.PodStore, proxier proxy.Proxier, k8sCl
 	nodeUID := string(node.UID)
 	klog.InfoS("Retrieved this Node's UID from K8s", "nodeName", nodeName, "nodeUID", nodeUID)
 
-	var exp exporter.Interface
+	var staticConsumerRes = createDestinationResFromOptions(o)
+
+	fe := &FlowExporter{
+		k8sClient: k8sClient,
+		crdClient: crdClient,
+
+		staticConsumerRes:   staticConsumerRes,
+		v4Enabled:           v4Enabled,
+		v6Enabled:           v6Enabled,
+		nodeRouteController: nodeRouteController,
+		isNetworkPolicyOnly: trafficEncapMode.IsNetworkPolicyOnly(),
+		egressQuerier:       egressQuerier,
+		podStore:            podStore,
+		l7Listener:          l7Listener,
+		nodeName:            nodeName,
+		nodeUID:             nodeUID,
+		obsDomainID:         obsDomainID,
+		destinationInformer: destinationInformer,
+		destinationLister:   destinationInformer.Lister(),
+		store:               ctStore,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
+			workqueue.NewTypedItemExponentialFailureRateLimiter[string](minRetryDelay, maxRetryDelay),
+			workqueue.TypedRateLimitingQueueConfig[string]{
+				Name: "flowexporterdestination",
+			},
+		),
+		addConsumerCh: make(chan *api.FlowExporterDestination),
+		rmConsumerCh:  make(chan string),
```

Comment:
do we need the extra indirection / complexity? You already use a workqueue so unless it takes seconds to add / remove a destination, I don't think you need the extra queues / channels. And those are unbuffered anyway. Maybe I'm wrong about what the intent was here.